src/devices/src/virtio/net/persist.rs

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Defines the structures needed for saving/restoring net devices.
+
+use std::sync::atomic::AtomicUsize;
+use std::sync::Arc;
+
+use rate_limiter::{persist::RateLimiterState, RateLimiter};
+use snapshot::Persist;
+use versionize::{VersionMap, Versionize, VersionizeResult};
+use versionize_derive::Versionize;
+use vm_memory::GuestMemoryMmap;
+
+use super::*;
+
+use crate::virtio::persist::VirtioDeviceState;
+use crate::virtio::{DeviceState, Queue};
+
+#[derive(Versionize)]
+pub struct NetConfigSpace {
+    guest_mac: Vec<u8>,
+}
+
+#[derive(Versionize)]
+pub struct NetState {
+    id: String,
+    tap_if_name: String,
+    rx_rate_limiter_state: RateLimiterState,
+    tx_rate_limiter_state: RateLimiterState,
+    allow_mmds_requests: bool,
```

Comment:
We also need to save and restore `mmds_ns: Option<MmdsNetworkStack>` since `MmdsNetworkStack` has state of its own.

`allow_mmds_requests` could be dropped from `NetState` and be inferred from `mmds_ns: Option<MmdsNetworkStack>`.

Reply:
Done, now we save the state of the MmdsNetworkStack.